cdc/redo/writer/writer_test.go

```diff
@@ -751,7 +751,7 @@ func TestWriterRedoGC(t *testing.T) {
 			cfg:       cfg,
 		}
 		go writer.runGC(context.Background())
-		time.Sleep(time.Duration(defaultGCIntervalInMs+1) * time.Millisecond)
+		time.Sleep(time.Duration(defaultGCIntervalInMs+10) * time.Millisecond)
```

Comment:
Prefer to add a callback function in `mockWriter.On("GC", mock.Anything).Return(nil)`, wait for the condition to be true and check GC has ran as expected.
such as
```diff
+               gcRunCounter := atomic.NewInt32(0)
                if tt.args.isRunning {
-                       mockWriter.On("GC", mock.Anything).Return(nil)
+                       mockWriter.On("GC", mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+                               gcRunCounter.Inc()
+                       })
                }
                writer := LogWriter{
                        rowWriter: mockWriter,
@@ -751,13 +755,18 @@ func TestWriterRedoGC(t *testing.T) {
                        cfg:       cfg,
                }
                go writer.runGC(context.Background())
-               time.Sleep(time.Duration(defaultGCIntervalInMs+1) * time.Millisecond)

                writer.Close()
                mockWriter.AssertNumberOfCalls(t, "Close", 2)

                if tt.args.isRunning {
-                       mockWriter.AssertCalled(t, "GC", mock.Anything)
+                       require.Eventually(t, func() bool {
+                               if gcRunCounter.Load() != int32(2) {
+                                       return false
+                               }
+                               mockWriter.AssertCalled(t, "GC", mock.Anything)
+                               return true
+                       }, time.Second, time.Millisecond*1)
```